admin/src/components/Wysiwyg/WysiwygNav.tsx

```diff
@@ -0,0 +1,266 @@
+/*
+ *
+ * WYSIWYG Nav
+ *
+ */
+import React, { useRef, useState } from 'react';
+import { useIntl } from 'react-intl';
+//@ts-ignore
+import { FocusTrap } from '@strapi/design-system/FocusTrap';
+//@ts-ignore
+import { Box } from '@strapi/design-system/Box';
+//@ts-ignore
+import { Button } from '@strapi/design-system/Button';
+//@ts-ignore
+import { IconButtonGroup } from '@strapi/design-system/IconButton';
+//@ts-ignore
+import { Option, Select } from '@strapi/design-system/Select';
+//@ts-ignore
+import { Popover } from '@strapi/design-system/Popover';
+//@ts-ignore
+import { Flex } from '@strapi/design-system/Flex';
+//@ts-ignore
+import Bold from '@strapi/icons/Bold';
+//@ts-ignore
+import Italic from '@strapi/icons/Italic';
+//@ts-ignore
+import Underline from '@strapi/icons/Underline';
+//@ts-ignore
+import StrikeThrough from '@strapi/icons/StrikeThrough';
+//@ts-ignore
+import BulletList from '@strapi/icons/BulletList';
+//@ts-ignore
+import NumberList from '@strapi/icons/NumberList';
+//@ts-ignore
+import Code from '@strapi/icons/Code';
+//@ts-ignore
+import Image from '@strapi/icons/Picture';
+//@ts-ignore
+import Link from '@strapi/icons/Link';
+//@ts-ignore
+import Quote from '@strapi/icons/Quote';
+//@ts-ignore
+import More from '@strapi/icons/More';
+import {
+  MainButtons,
+  CustomIconButton,
+  MoreButton,
+  IconButtonGroupMargin,
+  CustomLinkIconButton,
+} from './WysiwygStyles';
+
+type WysiwygNavProps = {
+  disabled?: boolean,
+  editorRef: any,
+  isExpandMode: boolean,
+  isPreviewMode: boolean,
+  onActionClick: (value:string,currentEditorRef:string,togglePopover?:any ) => void,
+  onToggleMediaLib: () => void,
+  onTogglePreviewMode: undefined | (() => void)
+}
+
+const WysiwygNav: React.FC<WysiwygNavProps> = ({
+  disabled,
+  editorRef,
+  isExpandMode,
+  isPreviewMode,
+  onActionClick,
+  onToggleMediaLib,
+  onTogglePreviewMode,
+}) => {
+  const [visiblePopover, setVisiblePopover] = useState<boolean>(false);
+  const { formatMessage } = useIntl();
+  const selectPlaceholder = formatMessage({
+    id: 'components.Wysiwyg.selectOptions.title',
+    defaultMessage: 'Add a title',
+  });
+  const buttonMoreRef = useRef();
+
+  const handleTogglePopover = () => {
+    setVisiblePopover(prev => !prev);
+  };
+
+  if (disabled || isPreviewMode) {
+    return (
+      <Box padding={2} background="neutral100">
```

Comment:
Consider extracting JSX of disabled/preview mode and regular mode into two separate components. IMO it will improve readability.

Reply:
wysiwyg should stays as is, in case of changes from Strapi side